tests/contrib/io/matplotlib/test_matplotlib_writer.py

```diff
@@ -0,0 +1,122 @@
+        # generate a plot
+        # pylint: disable=unsubscriptable-object,useless-suppression
+        plt.plot(np.random.rand(1, 5)[0], np.random.rand(1, 5)[0])
+
+        # write and compare
+        trusted_filepath = tmp_path / "image_we_expect.png"
+        plt.savefig(str(trusted_filepath))
+
+        experimental_filepath = tmp_path / "image_we_write.png"
+        plot_writer = MatplotlibWriter(filepath=str(experimental_filepath))
+        plot_writer.save(plt)
+        plt.close()
+
+        assert trusted_filepath.read_bytes() == experimental_filepath.read_bytes()
+
+    def test_list_images(self, tmp_path):
+        # generate plots
+        plots = list()
+        for index in range(5):
+            plots.append(plt.figure())
+            # pylint: disable=unsubscriptable-object,useless-suppression
+            plt.plot(np.random.rand(1, 5)[0], np.random.rand(1, 5)[0])
+
+        experimental_filepath = tmp_path / "list_images"
+        plot_writer = MatplotlibWriter(filepath=str(experimental_filepath))
+        plot_writer.save(plots)
+
+        # write and compare
+        for index, plot in enumerate(plots):
+            trusted_filepath = tmp_path / "image_we_expect_{}.png".format(str(index))
+            plot.savefig(str(trusted_filepath))
+
+            full_experimental_filepath = experimental_filepath / "{}.png".format(
+                str(index)
+            )
+            assert (
+                trusted_filepath.read_bytes() == full_experimental_filepath.read_bytes()
+            )
+
+    def test_dict_images(self, tmp_path):
+        plots = dict()
+        # generate plots
+        for index in ["boo", "far"]:
+            filename = "{}.png".format(index)
+
+            plots[filename] = plt.figure()
+            # pylint: disable=unsubscriptable-object,useless-suppression
+            plt.plot(np.random.rand(1, 5)[0], np.random.rand(1, 5)[0])
+
+        plot_writer = MatplotlibWriter(filepath=str(tmp_path / "dict_images"))
+
+        plot_writer.save(plots)
+
+        # write and compare
+        for filename, plot in plots.items():
+            trusted_filepath = (
+                tmp_path / "dict_images" / filename.replace(".png", "_trusted.png")
+            )
+            plot.savefig(str(trusted_filepath))
+            experimental_filepath = tmp_path / "dict_images" / filename
+
+            assert trusted_filepath.read_bytes() == experimental_filepath.read_bytes()
+
+    def test_load_fail(self, tmp_path):
+        plot_writer = MatplotlibWriter(filepath=str(tmp_path / "some_path"))
+
+        expected_load_error = "Loading not supported for MatplotlibWriter"
+        with pytest.raises(DataSetError, match=expected_load_error):
+            plot_writer.load()
+
+    def test_exists(self, tmp_path):
+        plot_object = plt.figure()
+        # pylint: disable=unsubscriptable-object,useless-suppression
+        plt.plot(np.random.rand(1, 5)[0], np.random.rand(1, 5)[0])
+        plt.close()
```

Comment:
Do we need to add this in the above tests too (test_save_list, test_save_dict)? I don't know if/what bad things happen.

Reply:
Good catch! `_exists()` does behave unexpectedly (returning False even after `.save` was called) with list and dict now, so I will fix it :)